src/systems/idle-detector.js

```diff
@@ -17,48 +17,45 @@ const BASIC_ACTIVITY_PATHS = [
   paths.actions.angularVelocity
 ];
 
-AFRAME.registerSystem("idle-detector", {
-  init() {
-    this.resetTimeout = this.resetTimeout.bind(this);
-    this.idleTimeout = null;
-    this.lastInputCheck = 0;
+let idleTimeout = null;
+const lastInputCheck = 0;
 
-    const events = ["click", "pointerdown", "touchstart", "keyup"];
+const onIdleTimeout = () => {
+  window.dispatchEvent(new CustomEvent("idle_detected"));
+};
 
-    for (const event of events) {
-      window.addEventListener(event, this.resetTimeout);
-    }
+const resetTimeout = () => {
+  if (idleTimeout) clearTimeout(idleTimeout);
+  idleTimeout = setTimeout(onIdleTimeout, IDLE_TIMEOUT_MS);
+  window.dispatchEvent(new CustomEvent("activity_detected"));
+};
 
-    this.resetTimeout();
-  },
-  resetTimeout() {
-    if (this.idleTimeout) clearTimeout(this.idleTimeout);
-    this.idleTimeout = setTimeout(this.onIdleTimeout, IDLE_TIMEOUT_MS);
-    window.dispatchEvent(new CustomEvent("activity_detected"));
-  },
-  onIdleTimeout() {
-    window.dispatchEvent(new CustomEvent("idle_detected"));
-  },
-  tick(time) {
-    if (time - this.lastInputCheck < INPUT_CHECK_INTERVAL_MS) return;
+for (const event of ["click", "pointerdown", "touchstart", "keyup"]) {
+  window.addEventListener(event, resetTimeout);
+}
 
-    const userinput = this.el.systems.userinput;
+resetTimeout();
 
-    let basicActivity = false;
-    for (const activityPath of BASIC_ACTIVITY_PATHS) {
-      basicActivity = basicActivity || !!userinput.get(activityPath);
-    }
+export const idleDetectSystem = time => {
+  // TODO: When should lastInputCheck be updated?
+  if (time - lastInputCheck < INPUT_CHECK_INTERVAL_MS) return;
```

Comment:
`lastInputCheck` doesn't seem to be updated anywhere. It seems a bug but I kept it as is so far because I'm not sure when it should be updated and what this check is really for.

Reply:
I don't see `lastInputCheck` being set anywhere in the current code, so it does seem like a bug. We should use porting systems as an opportunity to rethink the problem from the beginning. I would try to avoid the temptation of just blindly porting things forward. We should be aggressive about cutting things that don't make sense, or even cutting things that might be features but add way more complexity than value.